### DISCUSSION TITLE
Tick without overlapping

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#54018] [Q&A]
currently the `Octane::tick()` overlapping the previous tick, how do i prevent that ?

**Comment by zigzagdev** [comment]
@foremtehan 

Hi ! I'm Masatoshi.

To stop overlapping the previous tick, need not to start next tick. How about to write down this below into code ?

1. Implement a boolean flag to denote the running state of a tick.
2. Prior to initiating a new tick, interrogate the flag to ascertain if a tick is currently in progress. Only commence a new tick if the flag indicates that no tick is active.
3. Upon the successful conclusion of a tick, reset the flag to its initial state.

I hope this will helps to your question !! 🙌
